Sources/Extensions/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -135,6 +135,22 @@ public extension Sequence {
         })
     }
 
+    /// SwifterSwift: Get the only element based on optional condition.
+    ///
+    ///     [4].single() -> 4
+    ///     [2, 4].single() -> nil
+    ///     [1, 4, 7].single(where: {$0 % 2 == 0}) -> 4
+    ///     [2, 2, 4, 7].single(where: {$0 % 2 == 0}) -> nil
+    ///
+    /// - Parameter condition: condition to evaluate each element against (default is `{_ in true}`).
+    /// - Returns: The only element in the array matching the specified condition. If there are more matching elements, nil is returned. (optional)
+    public func single(where condition: ((Element) throws -> Bool) = { _ in true }) -> Element? {
+        if let filtered = try? filter(condition),
```

Comment:
Calling `filter` will _always_ run over the entire sequence. I'd like to suggest manually running through the sequence so that it can break out of the loop early, if possible.
```Swift
func single(where condition: ((Element) throws -> Bool) = { _ in true }) -> Element? {
    var singleElement: Element?
    for element in self where (try? condition(element)) ?? false {
        guard singleElement == nil else {
            singleElement = nil
            break
        }
        singleElement = element
    }
    return singleElement
}
```

Reply:
Great idea 👍 
I changed it accordantly